pkg/kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go

```diff
@@ -555,8 +555,10 @@ func (m *rangefeedMuxer) restartActiveRangeFeed(
 // on this node connection.  If no error returned, registers stream
 // with this connection.  Otherwise, stream is not registered.
 func (c *muxStream) startRangeFeed(
-	streamID int64, stream *activeMuxRangeFeed, req *kvpb.RangeFeedRequest,
+	streamID int64, stream *activeMuxRangeFeed, req *kvpb.RangeFeedRequest, beforeSend func(),
 ) error {
+	c.streams.Store(streamID, unsafe.Pointer(stream))
```

Comment:
Wouldn't that be a problem if we do `startRangeFeed()` while we are closing?
We will add stream to streams, realize that `c.mu.closed == true` return an error but keep stream in the map.
Outer loop will keep trying, but `close()` loop will also add this feed to its `toRestart` result to trigger restarts.

Reply:
Good points.  Addressed (I hope); added with large comment.